WordPress/Classes/ViewRelated/Comments/CommentDetailInfoViewModel.swift

```diff
@@ -0,0 +1,79 @@
+import Foundation
+
+protocol CommentDetailInfoViewModelInputs {
+    func didSelectItem(at index: Int)
+}
+
+protocol CommentDetailInfoViewModelOutputs {
+    func fetchUserDetails() -> [CommentDetailInfoUserDetails]
+}
+
+typealias CommentDetailInfoViewModelType = CommentDetailInfoViewModelInputs & CommentDetailInfoViewModelOutputs
+
+struct CommentDetailInfoUserDetails {
+    let title: String
+    let description: String
+}
+
+final class CommentDetailInfoViewModel: CommentDetailInfoViewModelType {
+    private let url: URL?
+    private let urlToDisplay: String?
+    private let email: String?
+    private let ipAddress: String?
+    private let isAdmin: Bool
+
+    weak var view: CommentDetailInfoView?
+
+    init(url: URL?, urlToDisplay: String?, email: String?, ipAddress: String?, isAdmin: Bool) {
+        self.url = url
+        self.urlToDisplay = urlToDisplay
+        self.email = email
+        self.ipAddress = ipAddress
+        self.isAdmin = isAdmin
+    }
+
+    func fetchUserDetails() -> [CommentDetailInfoUserDetails] {
```

Comment:
Not a huge deal but it seems like this could be calculated during the `init` since all the variables used in this function don't change.

Reply:
Yeah probably a good idea to run this once. I'll make it a variable and calculate it on init.